std/algebra/emulated/fields_bn254/hints.go

```diff
@@ -186,34 +187,67 @@ func divE6By6Hint(nativeMod *big.Int, nativeInputs, nativeOutputs []*big.Int) er
 func inverseE12Hint(nativeMod *big.Int, nativeInputs, nativeOutputs []*big.Int) error {
 	return emulated.UnwrapHint(nativeInputs, nativeOutputs,
 		func(mod *big.Int, inputs, outputs []*big.Int) error {
+			var d [12]big.Int
```

Comment:
I would refactor mapping into separate function and then call here and in divE12Hint. Same for the reverse mapping.